runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -422,13 +424,33 @@ public void run() {
     commitThread.setName("CommitThread");
 
     this.publishCounters = publishCounters;
+    this.clientId = clientIdGenerator.nextLong();
     this.windmillServer = options.getWindmillServerStub();
     this.windmillServer.setProcessHeartbeatResponses(this::handleHeartbeatResponses);
+    Duration maxBackoff =
+        !options.isEnableStreamingEngine() && options.getLocalWindmillHostport() != null
+            ? GrpcWindmillServer.LOCALHOST_BACKOFF
+            : GrpcWindmillServer.MAX_BACKOFF;
+    GrpcWindmillStreamFactory windmillStreamFactory =
```

Comment:
seems odd to create this here if the server isn't grpc server.

Can we just pass in somethign generic like JobHeader to windmillServer.start?